### ISSUE TITLE
Linux kernels 5.9..5.10: Cannot read all data in CH 22.7 ( pread_threadpool )

### ISSUE THREAD

**Issue by den-crane** [ClickHouse/ClickHouse#39753]
I created the issue to ease a search of a solution for others.

solution:

```xml
cat /etc/clickhouse-server/users.d/local_filesystem_read_method.xml
<?xml version="1.0" ?>
<clickhouse>
    <profiles>
        <default>
            <local_filesystem_read_method>pread</local_filesystem_read_method>
        </default>
    </profiles>
</clickhouse>
```

---------------------------

Cannot read from file 37, errno: 1, strerror: Operation not permitted: (while reading column event_date): (while reading from part /var/lib/clickhouse/store/0c4/0c486c57-095c-4343-b0b6-326807f9fc21/202207_494331_555280_2626/ from mark 0 with max_rows_to_read = 65505): While executing MergeTreeThread. (CANNOT_READ_FROM_FILE_DESCRIPTOR) (version 22.7.1.2484 (official build))

CentOS Linux release 7.9.2009 (Core)
3.10.0-1160.71.1.el7.x86_64 #1 SMP Tue Jun 28 15:37:28 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

---------------------------

Cannot read all data. Bytes read: 262. Bytes expected: 268.: (while reading column date): (while reading from part /var/lib/clickhouse-server/data/myapp/events_shard/202206_22_22_0/ from mark 48 with max_rows_to_read = 65505): While executing MergeTreeThread. (CANNOT_READ_ALL_DATA)

Debian GNU/Linux 11 (bullseye)
5.10.0-16-cloud-arm64 #1 SMP Debian 5.10.127-1 (2022-06-30) aarch64 GNU/Linux

---------------------------

Cannot read all data. Bytes read: 26613. Bytes expected: 47279.: (while reading column fare_amount): (while reading from part /var/lib/clickhouse/store/a2e/a2ee6c32-4979-415d-9ca0-727d9aa0613c/201503_33_38_1/ from mark 24 with max_rows_to_read = 65505): While executing MergeTreeThread. (CANNOT_READ_ALL_DATA)" while running the following query select min(fare_amount) from yellow_taxi for 146 M lines

Ubuntu 20.04.4 LTS
5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

---------------------------

Cannot read all data. Bytes read: 46098. Bytes expected: 62601.: (while reading column imsi): (while reading from
part /data/clickhouse/store/6fe/6fe42505-d6c7-4d87-bb8d-05d49a90c2cc/all_1_160_3/ from mark 0 with max_rows_to_read = 61272): While
executing MergeTreeThread. (CANNOT_READ_ALL_DATA) (version 22.7.1.2033 (official build))

---------------------------

Cannot read all data. Bytes read: 44772. Bytes expected: 47817.: (while reading column STAYNUM): (while reading from part /var/lib/clickhouse/store/40d/40dd58ff-55eb-4877-922d-f797f1db9731/all_7_12_1/ from mark 110 with max_rows_to_read = 24576): While executing MergeTreeThread. (CANNOT_READ_ALL_DATA) (version 22.7.1.2484 (official build))

---------------------------

**Comment by alexey-milovidov** [comment]
Old Linux Kernel versions have a bug in the preadv2 system call.
We will check the version and disable this code path if ClickHouse is run on an old Linux kernel.
The fix is here: #39800

**Comment by alexey-milovidov** [comment]
Let's collect ideas on how we can randomize Linux Kernel in our CI infrastructure.
I found this page: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/UserProvidedKernels.html
but it looks very difficult.
